src/core/public/doc_links/doc_links_service.mock.ts

```diff
@@ -18,7 +18,14 @@
  */
 import type { PublicMethodsOf } from '@kbn/utility-types';
 import { injectedMetadataServiceMock } from '../injected_metadata/injected_metadata_service.mock';
-import { DocLinksService, DocLinksStart } from './doc_links_service';
+import { DocLinksService, DocLinksSetup, DocLinksStart } from './doc_links_service';
+
+const createSetupContractMock = (): DocLinksSetup => {
+  // This service is so simple that we actually use the real implementation
+  const injectedMetadata = injectedMetadataServiceMock.createSetupContract();
+  injectedMetadata.getKibanaBranch.mockReturnValue('mocked-test-branch');
+  return new DocLinksService().setup({ injectedMetadata });
+};
```

Comment:
NIT: could change `createStartContractMock` implem to `return createSetupContractMock()`